### PR TITLE
Create tournament models

### PR DESCRIPTION
**Pull request by pczarn**
This change is [Reviewable](https://reviewable.io/reviews/pczarn/gameroom-task/5)

### CODE REVIEW

**Comment by jotka87** [comment]
Reviewed 22 of 25 files at r1.
Review status: 22 of 25 files reviewed at latest revision, 6 unresolved discussions.

---

_[app/models/tournament.rb, line 1 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRn7A2LPSPS3BxCHYY2:-KRn7A2Mq5VI1cNn912V:b-kf4924) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/app/models/tournament.rb#L1)):_

> ``` Ruby
> class Tournament < ApplicationRecord
> ```

please add a comment explaining what's the implemented type of tournament

---

_[db/migrate/20160905113939_create_matches.rb, line 4 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRn82Jdjwg2xrcEpGky:-KRn82Jdjwg2xrcEpGkz:bzgybhe) ([raw file](https://github.com/pczarn/gameroom-task/blob/52f965617c7d11d4bcaeed4f4a90cdcf286d3464/db/migrate/20160905113939_create_matches.rb#L4)):_

> ``` Ruby
>   def change
>     create_table :matches do |t|
>       t.datetime :played_at, null: false
> ```

that kind of change should go in a separate migration, you shouldn't modify the existed migration, here's why:
- if a migration was already merged to a master it cannot be modified
- as long as migration was created in a branch that is under development, there is a code review process in progress etc. it can be modified
- current case is the situation when business logic was changed and that happen pretty often

So, to sum up - matches migration is freezed, it cannot be modified but tournaments migration can be modified (because it was added only in this branch). Do you understand the difference?

---

_[db/migrate/20160915075805_create_tournaments.rb, line 7 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRn9v9xiOhafn_HS30y:-KRn9v9yrkvQ5WAn2hKB:b-d73jk4) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/db/migrate/20160915075805_create_tournaments.rb#L7)):_

> ``` Ruby
>       t.references :game, foreign_key: true, null: false
>       t.integer :status, null: false, default: 0
>       t.integer :number_of_teams, null: false
> ```

maybe some default value?

---

_[db/migrate/20160915120436_create_team_tournaments.rb, line 6 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRnEazQu2c9tdH8l0ms:-KRnEazQu2c9tdH8l0mt:b-q068iz) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/db/migrate/20160915120436_create_team_tournaments.rb#L6)):_

> ``` Ruby
>       t.references :team, foreign_key: true, null: false
>       t.references :tournament, foreign_key: true, null: false
>       t.integer :size_limit
> ```

maybe `max_teams_count`? or `teams_limit`?

---

_[spec/models/match_spec.rb, line 79 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRnHIBbMxnyRu8fAVB_:-KRnHIBcd0SjfQZtH6vv:b3bzfd6) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/spec/models/match_spec.rb#L79)):_

> ``` Ruby
>       it "belongs to a round" do
>         expect(match.round).to be_a(Round)
>       end
> ```

those two tests also don't make much sense, you are simply building a match with round and without and you're testing if it was assigned or not.

---

_[spec/models/tournament_spec.rb, line 117 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRnELEXP2LGFs7ZGnFr:-KRnELEXP2LGFs7ZGnFs:b-n4lx1m) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/spec/models/tournament_spec.rb#L117)):_

> ``` Ruby
>   describe "#image" do
>     it "is present" do
>       expect(tournament.image).to be_present
> ```

this test doesn't make much sense, you're testing that image was assigned in a factory basically

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by pczarn** [comment]
Review status: 22 of 25 files reviewed at latest revision, 7 unresolved discussions.

---

_[app/models/user.rb, line 12 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRnSzURkwoxVX9aJ0Zj:-KRnSzUS_1AUz7geOL7p:b-8g9zc6) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/app/models/user.rb#L12)):_

> ``` Ruby
>   has_many :teams, through: :user_teams
>   has_many :team_tournaments
>   has_many :tournaments, through: :team_tournaments
> ```

should I change the `tournaments` name? A user that creates tournaments will have many tournaments as their owner.

---

_[db/migrate/20160905113939_create_matches.rb, line 4 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRn82Jdjwg2xrcEpGky:-KRnOWdwz1eyAP5pJSxf:bnenvj1) ([raw file](https://github.com/pczarn/gameroom-task/blob/52f965617c7d11d4bcaeed4f4a90cdcf286d3464/db/migrate/20160905113939_create_matches.rb#L4)):_
<details><summary>

<i>Previously, jotka87 (Kamila Straus) wrote…</i></summary>



> that kind of change should go in a separate migration, you shouldn't modify the existed migration, here's why:
> - if a migration was already merged to a master it cannot be modified
> - as long as migration was created in a branch that is under development, there is a code review process in progress etc. it can be modified
> - current case is the situation when business logic was changed and that happen pretty often
> 
> So, to sum up - matches migration is freezed, it cannot be modified but tournaments migration can be modified (because it was added only in this branch). Do you understand the difference?
> </details>
> Now I see the difference, I was just used to doing rollbacks.

---

_[db/migrate/20160915075805_create_tournaments.rb, line 7 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRn9v9xiOhafn_HS30y:-KRnRv_Tr0LFyFL0jpC2:b-e0pm0t) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/db/migrate/20160915075805_create_tournaments.rb#L7)):_
<details><summary>

<i>Previously, jotka87 (Kamila Straus) wrote…</i></summary>



> maybe some default value?
> </details>
> Better not, because the person that creates a tournament should choose its size, say 2, 4, or 8 etc.

---

_[db/migrate/20160915120436_create_team_tournaments.rb, line 6 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRnEazQu2c9tdH8l0ms:-KRnNykQ2M_MPIUuhAJ9:b-cxpuwr) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/db/migrate/20160915120436_create_team_tournaments.rb#L6)):_
<details><summary>

<i>Previously, jotka87 (Kamila Straus) wrote…</i></summary>



> maybe `max_teams_count`? or `teams_limit`?
> </details>
> The size limit is per team in a tournament, so maybe `team_size_limit`. Ok, I'm changing it. However, I don't use this field in any way yet.

---

_[spec/models/match_spec.rb, line 79 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRnHIBbMxnyRu8fAVB_:-KRnSSAfhfPO3d4Tpk65:b-yp7jw) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/spec/models/match_spec.rb#L79)):_
<details><summary>

<i>Previously, jotka87 (Kamila Straus) wrote…</i></summary>



> those two tests also don't make much sense, you are simply building a match with round and without and you're testing if it was assigned or not.
> </details>
> It looks like it's useless

---

_[spec/models/tournament_spec.rb, line 117 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRnELEXP2LGFs7ZGnFr:-KRnSIB672-lowCZ1Nm7:b-jata6m) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/spec/models/tournament_spec.rb#L117)):_
<details><summary>

<i>Previously, jotka87 (Kamila Straus) wrote…</i></summary>



> this test doesn't make much sense, you're testing that image was assigned in a factory basically
> </details>
> How can I test that such field is available? Are there other tests I can write for it?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by pczarn** [comment]
Review status: 12 of 23 files reviewed at latest revision, 8 unresolved discussions.

---

_[app/models/tournament.rb, line 17 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0Oc3CDtBOvZLCbj4l:-KS0MECc5DPDBDh8bc-_:b-6ufzhq) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/app/models/tournament.rb#L17)):_

> ``` Ruby
>   }
> 
>   mount_uploader :image, ImageUploader
> ```

Can I reuse the same uploader in two models? I think so.

---

_[app/models/tournament.rb, line 53 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0OW1hyJB7GRMujpw4:-KS0OW1iVxBjrbulY1g0:bjanst6) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/app/models/tournament.rb#L53)):_

> ``` Ruby
>         end
>       end
>     end
> ```

Should I move this validation to the `Match` model?

---

_[db/migrate/20160915120436_create_team_tournaments.rb, line 3 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0MyMY0a4A8z79NLb6:-KS0MyMZPo3ww_KVJ-XH:bezjipr) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/db/migrate/20160915120436_create_team_tournaments.rb#L3)):_

> ``` Ruby
> class CreateTeamTournaments < ActiveRecord::Migration[5.0]
>   def change
>     create_table :team_tournaments do |t|
> ```

Should I add an `id: false` parameter here so that the join table doesn't have an id column?

---

_[db/migrate/20160916083755_add_image_to_tournaments.rb, line 3 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0NCBMLtTwnVgiwVh5:-KS0NCBMLtTwnVgiwVh6:b-rpxo5n) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/db/migrate/20160916083755_add_image_to_tournaments.rb#L3)):_

> ``` Ruby
> class AddImageToTournaments < ActiveRecord::Migration[5.0]
>   def change
>     add_column :tournaments, :image, :string
> ```

Should I put it in the previous migration?

---

_[db/migrate/20160916101106_create_rounds.rb, line 11 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0NQKmE1Q4nbDF5YZY:-KS0NQKmE1Q4nbDF5YZZ:bggpjcf) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/db/migrate/20160916101106_create_rounds.rb#L11)):_

> ``` Ruby
> 
>     add_reference :matches, :round, foreign_key: true
>     change_column_null :matches, :played_at, true
> ```

Should I move changes to `matches` to a better named migration?

---

_[spec/controllers/matches_controller_spec.rb, line 64 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0N_801tXBGVHL-ml1:-KS0N_81yGsNYu8KTtrq:bdp3e8h) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/spec/controllers/matches_controller_spec.rb#L64)):_

> ``` Ruby
> 
>       it "renders the index" do
>         is_expected.to render_template "matches/index"
> ```

Should I be adding more tests for rendered templates?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by jotka87** [comment]
Reviewed 1 of 25 files at r1, 8 of 12 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

_[app/models/tournament.rb, line 17 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0Oc3CDtBOvZLCbj4l:-KS0PVnXf1Vf194ZkFJF:b-7wgyj6) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/app/models/tournament.rb#L17)):_
<details><summary>

<i>Previously, pczarn (Piotr Czarnecki) wrote…</i></summary>



> Can I reuse the same uploader in two models? I think so.
> </details>
> If the models will share the same uploader config then it's fine, but when it's different then it should be as a separate uploader

---

_[app/models/user.rb, line 13 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0KDb2W4Llts8-rblC:-KS0KDb3VsVHH0WwAXNK:b-gimwjv) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/app/models/user.rb#L13)):_

> ``` Ruby
>   has_many :team_tournaments
>   has_many :tournaments, through: :team_tournaments
>   has_many :created_tournaments, source: :tournament, class_name: Tournament
> ```

name is fine but you need to add also `foreign_key: :owner_id` to make that work

---

_[db/migrate/20160916101106_create_rounds.rb, line 11 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0GSS8ts4SGGwtx0Cj:-KS0GSS9QA9qFxr-upsE:bttg9ut) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/db/migrate/20160916101106_create_rounds.rb#L11)):_

> ``` Ruby
> 
>     add_reference :matches, :round, foreign_key: true
>     change_column_null :matches, :played_at, true
> ```

Basically that should go to a separate migration, not as a part of migration creating new table, because this is about modifying different table.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by wmaciejak** [comment]
_[app/models/tournament.rb, line 53 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0OW1hyJB7GRMujpw4:-KS0zi4HbP7-5A_HkygH:bj0b1rs) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/app/models/tournament.rb#L53)):_
<details><summary>

<i>Previously, pczarn (Piotr Czarnecki) wrote…</i></summary>



> Should I move this validation to the `Match` model?
> </details>
> Yes, I think, `Match` model is a better place for this validation, cause it's related to attributes of match

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by wmaciejak** [comment]
_[app/models/user.rb, line 12 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRnSzURkwoxVX9aJ0Zj:-KS1-QpFdNEFtuVndYOp:b-ypph1t) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/app/models/user.rb#L12)):_
<details><summary>

<i>Previously, pczarn (Piotr Czarnecki) wrote…</i></summary>



> should I change the `tournaments` name? A user that creates tournaments will have many tournaments as their owner.
> </details>
> Please look to the [line 14](https://github.com/pczarn/gameroom-task/blob/3f098f6e6f0bd5337f11beab5adc7887300a5e1f/app/models/user.rb#L14)

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by wmaciejak** [comment]
_[spec/controllers/matches_controller_spec.rb, line 64 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0N_801tXBGVHL-ml1:-KS11hMQYNwFd-K1NnlA:b-nw9bnt) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/spec/controllers/matches_controller_spec.rb#L64)):_
<details><summary>

<i>Previously, pczarn (Piotr Czarnecki) wrote…</i></summary>



> Should I be adding more tests for rendered templates?
> </details>
> In my opinion, every template should have a separate test for rendering it. But if in this place you have only one template I think it's enought.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by wmaciejak** [comment]
Review status: 18 of 24 files reviewed at latest revision, 9 unresolved discussions.

---

_[db/migrate/20160915120436_create_team_tournaments.rb, line 3 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0MyMY0a4A8z79NLb6:-KS10Vg7n8bzV1QSOclq:b-g1g8ul) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/db/migrate/20160915120436_create_team_tournaments.rb#L3)):_
<details><summary>

<i>Previously, pczarn (Piotr Czarnecki) wrote…</i></summary>



> Should I add an `id: false` parameter here so that the join table doesn't have an id column?
> </details>
> Yes, please add the parameter.

---

_[db/migrate/20160916083755_add_image_to_tournaments.rb, line 3 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0NCBMLtTwnVgiwVh5:-KS12CC192FmVaoDGNXj:b-f10nmt) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/db/migrate/20160916083755_add_image_to_tournaments.rb#L3)):_
<details><summary>

<i>Previously, pczarn (Piotr Czarnecki) wrote…</i></summary>



> Should I put it in the previous migration?
> </details>
> In my opinion, you can merge this migration to `20160915075805_create_tournaments.rb` file, cause it's still related to this same feature.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by wmaciejak** [comment]
Reviewed 1 of 12 files at r2, 1 of 5 files at r4.
Review status: 19 of 24 files reviewed at latest revision, 9 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by pczarn** [comment]
Review status: 12 of 24 files reviewed at latest revision, 8 unresolved discussions.

---

_[app/models/tournament.rb, line 53 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0OW1hyJB7GRMujpw4:-KS18yKXbGckxmxI-TVG:bxnj45) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/app/models/tournament.rb#L53)):_
<details><summary>

<i>Previously, wojtekmaciej (Wojciech Maciejak) wrote…</i></summary>



> Yes, I think, `Match` model is a better place for this validation, cause it's related to attributes of match
> </details>
> Ok. I'm going to move the spec for this, too.

---

_[app/models/user.rb, line 12 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRnSzURkwoxVX9aJ0Zj:-KS0M_zv2yDlCRq1wSfa:bbfva0d) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/app/models/user.rb#L12)):_
<details><summary>

<i>Previously, wojtekmaciej (Wojciech Maciejak) wrote…</i></summary>



> Please look to the [line 14](https://github.com/pczarn/gameroom-task/blob/3f098f6e6f0bd5337f11beab5adc7887300a5e1f/app/models/user.rb#L14)
> </details>
> I meant to say that when I give one thing a name like `owned_tournaments` instead of `tournaments`, I should rename the other `tournaments` thing too.

---

_[db/migrate/20160915120436_create_team_tournaments.rb, line 3 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0MyMY0a4A8z79NLb6:-KS1FaYr7UaHst-kBnJw:bkdkwml) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/db/migrate/20160915120436_create_team_tournaments.rb#L3)):_
<details><summary>

<i>Previously, wojtekmaciej (Wojciech Maciejak) wrote…</i></summary>



> Yes, please add the parameter.
> </details>
> Unfortunately validations don't work without an id column. Multi-column primary keys aren't easy to add.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by wmaciejak** [comment]
_[app/models/user.rb, line 12 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRnSzURkwoxVX9aJ0Zj:-KS1UxtD6-fTVLa1DHSA:biza4be) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/app/models/user.rb#L12)):_
<details><summary>

<i>Previously, pczarn (Piotr Czarnecki) wrote…</i></summary>



> I meant to say that when I give one thing a name like `owned_tournaments` instead of `tournaments`, I should rename the other `tournaments` thing too.
> </details>
> Ahh, okay, for me if the user is an owner of the tournament it should be named exactly what it is, for example: user can be an owner of some tournaments, and can be a player in some tournaments too. It can be confused if person looks first time to code. 
> cc @jotka87, if it's not a problem please tell wdyt :)

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by wmaciejak** [comment]
_[db/migrate/20160915120436_create_team_tournaments.rb, line 3 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0MyMY0a4A8z79NLb6:-KS1WB0OY8l984GLlUdj:bixged) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/db/migrate/20160915120436_create_team_tournaments.rb#L3)):_
<details><summary>

<i>Previously, pczarn (Piotr Czarnecki) wrote…</i></summary>



> Unfortunately validations don't work without an id column. Multi-column primary keys aren't easy to add.
> </details>
> LGTM, we told about it. Rails are really stupid ;)

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by wmaciejak** [comment]
Reviewed 1 of 5 files at r4, 10 of 10 files at r5, 1 of 1 files at r6.
Review status: 22 of 24 files reviewed at latest revision, 6 unresolved discussions.

---

_[db/migrate/20160916101106_create_rounds.rb, line 11 at r3](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS0GSS8ts4SGGwtx0Cj:-KS1War9S_6IwfTRhfI4:b6nbv0c) ([raw file](https://github.com/pczarn/gameroom-task/blob/91e6907b51d49ff913873abc262231afea3d792e/db/migrate/20160916101106_create_rounds.rb#L11)):_
<details><summary>

<i>Previously, jotka87 (Kamila Straus) wrote…</i></summary>



> Basically that should go to a separate migration, not as a part of migration creating new table, because this is about modifying different table.
> </details>
> Resolved for me.

---

_[db/migrate/20160919095227_update_matches.rb, line 1 at r6](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS1Wm6zouYgwO-yTBHg:-KS1Wm6zouYgwO-yTBHh:b-d1i8pr) ([raw file](https://github.com/pczarn/gameroom-task/blob/40587862c4171319b5aaf606e74d9d2017d71360/db/migrate/20160919095227_update_matches.rb#L1)):_

> ``` Ruby
> class UpdateMatches < ActiveRecord::Migration[5.0]
> ```

Name of migration should tell what happens inside it. What do you think about `AddRoundForeignKeyToMatches` ?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by pczarn** [comment]
Review status: 22 of 24 files reviewed at latest revision, 6 unresolved discussions.

---

_[db/migrate/20160919095227_update_matches.rb, line 1 at r6](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KS1Wm6zouYgwO-yTBHg:-KS1gBaYG1ZKwemIqVmv:b-xx886s) ([raw file](https://github.com/pczarn/gameroom-task/blob/40587862c4171319b5aaf606e74d9d2017d71360/db/migrate/20160919095227_update_matches.rb#L1)):_
<details><summary>

<i>Previously, wojtekmaciej (Wojciech Maciejak) wrote…</i></summary>



> Name of migration should tell what happens inside it. What do you think about `AddRoundForeignKeyToMatches` ?
> </details>
> Then I'll make an additional migration because this migration does two things

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by jotka87** [comment]
Reviewed 1 of 5 files at r4, 1 of 10 files at r5, 4 of 4 files at r7.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

_[app/models/user.rb, line 12 at r1](https://reviewable.io:443/reviews/pczarn/gameroom-task/5#-KRnSzURkwoxVX9aJ0Zj:-KS0I9MiRuKdQIa0uQY4:b-gl22w2) ([raw file](https://github.com/pczarn/gameroom-task/blob/a896709db50381cc3c72f04482425fad2e54a2c0/app/models/user.rb#L12)):_
<details><summary>

<i>Previously, wojtekmaciej (Wojciech Maciejak) wrote…</i></summary>



> Ahh, okay, for me if the user is an owner of the tournament it should be named exactly what it is, for example: user can be an owner of some tournaments, and can be a player in some tournaments too. It can be confused if person looks first time to code. 
> cc @jotka87, if it's not a problem please tell wdyt :)
> </details>
> I guess that names:
> - `tournaments` for tournaments where user participates and 
> - `owned_tournaments` for tournaments that user created 

are ok.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by wmaciejak** [comment]
Reviewed 1 of 5 files at r4.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_

**Comment by wmaciejak** [comment]
Reviewed 1 of 4 files at r7.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pczarn/gameroom-task/5)_
